adapters/repos/db/vector/hnsw/metrics.go

```diff
@@ -193,6 +201,21 @@ func (m *Metrics) SetSize(size int) {
 	m.size.Set(float64(size))
 }
 
+func (m *Metrics) VectorInfo(dimensions int, segments int, metric string, pq bool) {
+	if !m.enabled {
+		return
+	}
+
+	m.info.DeletePartialMatch(prometheus.Labels{"class_name": m.className})
```

Comment:
Do we have to delete the gauge before calling Set?

Reply:
Yes as otherwise duplicate metrics can occur i.e. a metric with `pq=true` and `pq=false`.